glue/core/data.py

```diff
@@ -1619,13 +1619,13 @@ def compute_histogram(self, cids, weights=None, range=None, bins=None, log=None,
             return np.zeros(bins)
 
         if log is not None and log[0]:
-            xmin = np.log10(xmin)
-            xmax = np.log10(xmax)
+            xmin = np.log10(xmin) if xmin > 0 else np.log10(xmax) - 10
+            xmax = np.log10(xmax) if xmax > 0 else np.log10(xmin) + 10
```

Comment:
This might give strange results if both xmin and xmax are zero for some reason - I think it might be best to actually just return an array of NaN values if either xmin or xmax are negative rather than modify the limits. You can construct this array efficiently with:

```
broadcast_to(np.nan, bins)
```

could you update this here and below?